server/src/main/scala/com/cloudera/livy/server/client/ClientSessionServlet.scala

```diff
@@ -53,17 +60,47 @@ class ClientSessionServlet(sessionManager: SessionManager[ClientSession, CreateC
     }
   }
 
+  jpost[Any]("/:id/upload-jar") { req =>
```

Comment:
nit: the better way to ignore the body here is to do:

    jpost[Unit]("/:id/upload-jar") { _ =>

I probably should make the documentation better at some point... I'm not sure what's really happening when you put `[Any]` in the type parameter.